### DISCUSSION TITLE
Setting state from outside of a component, without using setState

### DISCUSSION THREAD

**Discussion by KevinMusgrave** [pmndrs/zustand#1462] [Q&A] [answered]
Here's a simple state and action:

```javascript
import create from "zustand";

export const useName = create((set) => ({
    name: null,
    setName: (name) => set({name})
}));
```

To use the `name` state outside of a component I can do:

```javascript
const name = useName.getState().name
```

My question is, if I want to use the `setName` action, is the following ok?

```javascript
const setName = useName.getState().setName
setName("Kevin")
```

I know that there is a `setState()` function, but I think that would require replicating the logic that already exists in `useName`.

**Comment by dai-shi** [comment, marked as answer]
> My question is, if I want to use the setName action, is the following ok?

Yes.

**Reply by KevinMusgrave** [reply]
Thanks!
